include/tables/FileProvenanceLogTable.h

```diff
@@ -160,6 +160,42 @@ typedef boost::heap::priority_queue<FileProvenanceRow, boost::heap::compare<File
 typedef std::vector <boost::optional<FileProvenancePK> > PKeys;
 typedef std::vector <FileProvenanceRow> Pq;
 
+class FProvCache {
```

Comment:
I guess this cache is only used in FileProvenanceElasticDataReader.cpp, right? If so, we should move it to FileProvenanceElasticDataReader.cpp

Reply:
but the table is also only used in FileProvenanceElasticDataReader... doesn't it make more sense for the Cache to stay with the table?